src/wam2layers/preprocessing/pressure_levels.py

```diff
@@ -0,0 +1,139 @@
+from wam2layers.preprocessing.utils import interpolate, sortby_ndarray
+
+
+import numpy as np
+import xarray as xr
+
+
+def extend_pressurelevels(
+    input: xr.Dataset,
+    surface: xr.Dataset,
+) -> tuple[xr.Dataset, xr.DataArray]:
+    """Extend the levels to top of atmosphere, surface and 2-layer boundary level.
+
+    The input data does not have data at the surface and
+    and the top of atmosphere. The input data is extended to
+    these edges.
+
+    For proper integration, extra layers have to be inserted
+    near the 2-layer boundary. This boundary level is hard
+    coded as a function of surface pressure and not customizable.
+
+    Args:
+        input_data: Dataset with the following variables:
+            q: Specific humidity at pressure levels
+            u: Eastward horizontal wind speed at pressure levels
+            v: Northward horizontal wind speed at pressure levels
+        surface_data: Dataset with the following variables:
+            ps: Air pressure at the surface
+            qs: Specific humidity at the surface
+            us: Eastward horizontal wind speed at the surface
+            vs: Northward horizontal wind speed at the surface
+
+    Returns:
+        Dataset with the following variables (interpolated to new pressure levels):
+            p, q, u, v
+        The pressure at the division between the two layers
+    """
+    level_ax: int = input["u"].get_axis_num("level")  # type: ignore
+
+    da_p = input["level"].broadcast_like(input["u"]) * 100  # Pa
+    da_p.attrs["units"] = "Pa"
+
+    p = da_p.to_numpy()
+    u = input["u"].to_numpy()
+    v = input["v"].to_numpy()
+    q = input["q"].to_numpy()
+
+    # Insert top of atmosphere values
+    # Assume wind at top same as values at lowest pressure, humidity at top = 0
+    p0 = p.argmin(axis=level_ax).flatten()[
+        0
+    ]  # index along level axis of minimum pressure
+    p = np.insert(p, 0, 0.0, axis=level_ax)
+    q = np.insert(q, 0, 0.0, axis=level_ax)
+    v = np.insert(v, 0, np.take(v, p0, axis=level_ax), axis=level_ax)
+    u = np.insert(u, 0, np.take(u, p0, axis=level_ax), axis=level_ax)
+
+    # Insert surface level values
+    p = np.insert(p, 0, surface["ps"], axis=level_ax)
+    q = np.insert(q, 0, surface["qs"], axis=level_ax)
+    v = np.insert(v, 0, surface["vs"], axis=level_ax)
+    u = np.insert(u, 0, surface["us"], axis=level_ax)
+
+    # Sort arrays by pressure (ascending)
+    u = sortby_ndarray(u, p, axis=level_ax)
+    v = sortby_ndarray(v, p, axis=level_ax)
+    q = sortby_ndarray(q, p, axis=level_ax)
+    p = sortby_ndarray(p, p, axis=level_ax)
+
+    # Insert boundary level values (at a dummy pressure value)
+    p_boundary = 0.72878581 * np.array(surface["ps"]) + 7438.803223
```

Comment:
Pass this into the function?

Reply:
I've made it configurable, so now we only need to pass the config to the function